contracts/MiniMeToken.sol

```diff
@@ -281,6 +312,98 @@ contract MiniMeToken is Controlled {
         return totalSupplyAt(block.number);
     }
 
+////////////////
+// Permit and transferWithAuthorization
+////////////////
+    function permit(
+        address _owner,
+        address _spender,
+        uint256 _value,
+        uint256 _deadline,
+        uint8 _v,
+        bytes32 _r,
+        bytes32 _s
+    ) public {
+        require(_deadline >= block.timestamp, "permit: AUTH_EXPIRED");
+
+        bytes32 encodeData = keccak256(
+            abi.encode(
+                PERMIT_TYPEHASH,
+                _owner,
+                _spender,
+                _value,
+                nonces[_owner]++,
+                _deadline
+            )
+        );
+        _validateSignedData(_owner, encodeData, _v, _r, _s);
+        _approve(_owner, _spender, _value);
+    }
+
+    function getDomainSeparator() public constant returns (bytes32) {
+        return keccak256(
+            abi.encode(
+                EIP712DOMAIN_HASH,
+                nameHash,
+                VERSION_HASH,
+                CHAINID,
+                address(this)
+            )
+        );
+    }
+
+    function _validateSignedData(
+        address _signer,
+        bytes32 _encodeData,
+        uint8 _v,
+        bytes32 _r,
+        bytes32 _s
+    ) internal constant {
+
+        bytes32 digest = keccak256(
+            abi.encodePacked("\x19\x01", getDomainSeparator(), _encodeData)
+        );
+        address recoveredAddress = ecrecover(digest, _v, _r, _s);
+        // Explicitly disallow authorizations for address(0) as ecrecover returns address(0) on malformed messages
+        require(
+            recoveredAddress != 0 && recoveredAddress == _signer,
+            "_validateSignedData: INVALID_SIGNATURE"
+        );
+    }
+
+    function transferWithAuthorization(
+        address _from,
+        address _to,
+        uint256 _value,
+        uint256 _validAfter,
+        uint256 _validBefore,
+        bytes32 _nonce,
+        uint8 _v,
+        bytes32 _r,
+        bytes32 _s
+    ) external {
+        require(block.timestamp > _validAfter, "transferWithAuthorization: AUTH_NOT_YET_VALID");
+        require(block.timestamp < _validBefore, "transferWithAuthorization: AUTH_EXPIRED");
+        require(!authorizationState[_from][_nonce], "transferWithAuthorization: AUTH_ALREADY_USED");
+
+        bytes32 encodeData = keccak256(
+            abi.encode(
+                TRANSFER_WITH_AUTHORIZATION_TYPEHASH,
+                _from,
+                _to,
+                _value,
+                _validAfter,
+                _validBefore,
+                _nonce
+            )
+        );
+        _validateSignedData(_from, encodeData, _v, _r, _s);
+
+        authorizationState[_from][_nonce] = true;
+        require(doTransfer(_from, _to, _value));
```

Comment:
We need to check if `transfersEnabled` is true before executing this.

Reply:
You are totally right, I just added the check